agent/taskresource/cgroup/cgroup_unsupported.go

```diff
@@ -113,12 +113,12 @@ func (c *CgroupResource) GetTerminalReason() string {
 
 // MarshalJSON marshals CgroupResource object
 func (c *CgroupResource) MarshalJSON() ([]byte, error) {
-	return nil, errors.New("unsupported platform")
```

Comment:
We'd want this to fail loudly on windows if this path got crossed, right?

Reply:
yeah. keeping the error and fixes the state file instead